src/dialogs/edittagdialog.cpp

```diff
@@ -1411,6 +1415,23 @@ void EditTagDialog::FetchTagSongChosen(const Song &original_song, const Song &ne
 
 }
 
+void EditTagDialog::FetchLyrics() {
+
+  Song song = data_[ui_->song_list->selectionModel()->selectedIndexes().first().row()].original_;
```

Comment:
This can be const reference.

Also, you need to check that `selectedIndexes` is not empty, otherwise this will crash.

Add `if (ui_->song_list->selectionModel()->selectedIndexes().isEmpty()) return;` before this line.

Reply:
I also think `current_` should be used here, not `original_`, so if the user changes the tags it uses the modified metadata.